### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by tribaeka**

### CODE REVIEW

File: app/core/component/course-control/course-control.controller.js [hunk 1]
```diff
@@ -2,21 +2,18 @@
...
         var $ctrl = this;
         $ctrl.showFromForAdd = false;
         $ctrl.applyFilter = function() {
             console.log('filter value = ' + $ctrl.inputFilterValue);
-            $rootScope.$broadcast(sendFiltersInputValueToCoursesFilterEvent, $ctrl.inputFilterValue);
+            $rootScope.$broadcast(eventsFactory.sendFiltersInputValueToCoursesFilterEvent, $ctrl.inputFilterValue);
             $ctrl.inputFilterValue = '';
         };
 
         $ctrl.toggleFromForAdd = function() {
```
Comment (kirillparamonov) [inline app/core/component/course-control/course-control.controller.js]: Please rename that too

File: app/core/component/course-list/course-list.scss [hunk 2]
```diff
@@ -1,11 +1,22 @@
 @import "../../../colors";
+@import "../../../mixins";
 
 .course-container {
   border-radius: 0.25rem;
 
+  .top-rated-course {
+    background: $lightBlueColor;
+  }
   .info {
```
Comment (kirillparamonov) [inline app/core/component/course-list/course-list.scss]: Please add an empty line

File: app/core/component/course-list/course-list.template.html [hunk 3]
```diff
@@ -1,36 +1,55 @@
...
-                </button>
+    <ul class="course-list">
+        <li class="course-item"
+            state-by-date="course"
+            ng-repeat="course in $ctrl.courses |
+                limitTo:$ctrl.coursePullSize |
+                filter: $ctrl.filterValue |
+                orderBy: $ctrl.creationDateToTime">
+            <div class="course-wrapper"
+                 ng-class="$ctrl.topRatedClassIsIt(course)">
```
Comment (kirillparamonov) [inline app/core/component/course-list/course-list.template.html]: isTopRated will be more readable

File: app/core/component/forms/forms.controller.js [hunk 4]
```diff
@@ -2,50 +2,42 @@
...
-        $rootScope, addCourseEvent, pushCourseToEditFormEvent,
-        editCourseEvent, toggleVisibilityFromForAddEvent
-    ) {
+    .controller('formsController', function($rootScope, courseService, eventsFactory) {
         // eslint-disable-next-line consistent-this,no-invalid-this
         var $ctrl = this;
 
-        $rootScope.$on(toggleVisibilityFromForAddEvent, function(event, data) {
+        $rootScope.$on(eventsFactory.toggleVisibilityFromForAddEvent, function(event, data) {
             $ctrl.showFromForAdd = data;
```
Comment (kirillparamonov) [inline app/core/component/forms/forms.controller.js]: Please rename it either

File: app/core/component/forms/forms.controller.js [hunk 5]
```diff
@@ -2,50 +2,42 @@
...
             $ctrl.showFromForAdd = !$ctrl.showFromForAdd;
         };
 
-        $rootScope.$on(pushCourseToEditFormEvent, function(event, data) {
-            $ctrl.selectedCourse = data;
-            $ctrl.courseTitleToEdit = data.title;
-            $ctrl.courseDescriptionToEdit = data.description;
-            $ctrl.courseUploadDateToEdit = data.uploadDate;
-            $ctrl.courseDurationToEdit = data.duration;
+        $rootScope.$on(eventsFactory.pushCourseToEditFormEvent, function(event, data) {
```
Comment (kirillparamonov) [inline app/core/component/forms/forms.controller.js]: pushCourseToEditFormEvent says that you should push Course:) 
so I believe that data should be renamed.

File: app/core/component/forms/forms.controller.js [hunk 6]
```diff
@@ -2,50 +2,42 @@
...
+            $ctrl.toEditCourse = {
+                title: data.title,
+                description: data.description,
+                creationDate: data.creationDate,
+                duration: data.duration,
+                topRated: data.topRated,
+                selectedCourse: data
+            };
             $ctrl.showFromForEdit = true;
```
Comment (kirillparamonov) [inline app/core/component/forms/forms.controller.js]: Please rename it too

File: app/core/component/course-list/course-list.scss [hunk 7]
```diff
@@ -1,11 +1,22 @@
...
     width: 85%;
     display: inline-block;
+    margin-left: 2rem;
+
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/component/course-list/course-list.scss]: removed

File: app/core/component/course-list/course-list.template.html [hunk 8]
```diff
@@ -1,36 +1,55 @@
...
+                <div class="info">
+                    <div class="flex-justify-between">
+                        <h3>
+                            {{course.title | uppercase}}
+                            <img src="img/star.svg"
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/component/course-list/course-list.template.html]: fixed

File: app/core/component/course-control/course-control.controller.js [hunk 9]
```diff
@@ -2,21 +2,18 @@
...
 angular
     .module('courseControl')
-    .controller('CourseControlController', function(
-        $rootScope, sendFiltersInputValueToCoursesFilterEvent,
-        toggleVisibilityFromForAddEvent
-    ) {
+    .controller('CourseControlController', function($rootScope, eventsFactory) {
         // eslint-disable-next-line consistent-this,no-invalid-this
         var $ctrl = this;
         $ctrl.showFromForAdd = false;
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/component/course-control/course-control.controller.js]: renamed

File: app/core/component/course-control/course-control.controller.js (hunk 1 above)
Comment (tribaeka) [reply app/core/component/course-control/course-control.controller.js]: renamed too + in all files

File: app/core/component/course-list/course-list.scss (hunk 2 above)
Comment (tribaeka) [reply app/core/component/course-list/course-list.scss]: fixed

File: app/core/component/forms/forms.controller.js (hunk 4 above)
Comment (tribaeka) [reply app/core/component/forms/forms.controller.js]: renamed

File: app/core/component/forms/forms.controller.js (hunk 6 above)
Comment (tribaeka) [reply app/core/component/forms/forms.controller.js]: renamed

File: app/core/service/course/course.service.js [hunk 10]
```diff
@@ -3,53 +3,36 @@
...
-            course.uploadTime = uploadDate.getTime();
+        function getCourses() {
+            return courses;
         }
 
-        function dateToDisplayDate(date) {
-            return ((date.getMonth() + 1) < 10 ? '0'+ (date.getMonth() + 1) : (date.getMonth() + 1)) + '.' +
-                (date.getDate() < 10 ? '0'+ date.getDate() : date.getDate()) + '.' +
-                date.getFullYear();
+        function editCourse(data) {
```
Comment (kirillparamonov) [inline app/core/service/course/course.service.js]: You have strange duplicated structure in you data argument:
image
As a result, course item in array, that is overridden by your  editCourse function, contains selectedCourse field.

File: app/core/component/forms/forms.controller.js (hunk 5 above)
Comment (tribaeka) [reply app/core/component/forms/forms.controller.js]: renamed;
courseExchangeWithEditForm is ok for it?

File: app/core/component/course-list/course-list.template.html (hunk 3 above)
Comment (tribaeka) [reply app/core/component/course-list/course-list.template.html]: renamed
ok but it return class name, it's okay for 'is...' function return not booleam?

File: app/core/service/course/course.service.js (hunk 10 above)
Comment (tribaeka) [reply app/core/service/course/course.service.js]: i used this field for indexOf() place to replace my old object before edit on new object after sending edit form

File: app/core/filter/durationFromMinutes/duration-from-minutes.filter.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+angular
+    .module('durationFromMinutes')
+    .filter('durationFromMinutes', function() {
+        return function(duration) {
+            return duration > 59 ?
+                duration % 60 === 0 ?
+                    Math.round(duration / 60) + 'h ' :
+                    Math.round(duration / 60) + 'h ' + duration % 60 + 'min' :
+                duration + 'min';
```
[the remark this answers is not in the document]
Comment (tribaeka) [reply app/core/filter/durationFromMinutes/duration-from-minutes.filter.js]: fixed

File: app/core/service/course/course.service.js (hunk 10 above)
Comment (tribaeka) [reply app/core/service/course/course.service.js]: fixed with using lodash

File: app/core/component/course-list/course-list.template.html (hunk 3 above)
Comment (kirillparamonov) [reply app/core/component/course-list/course-list.template.html]: According to your function implementation (if or nothing in return), I'd say, that it is more like boolean.

File: app/core/component/forms/forms.controller.js (hunk 5 above)
Comment (kirillparamonov) [reply app/core/component/forms/forms.controller.js]: sure, thanks

**Pull request merged**